digdag-cli/src/main/java/io/digdag/cli/CollectionPrinter.java

```diff
@@ -0,0 +1,94 @@
+package io.digdag.cli;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import com.fasterxml.jackson.datatype.guava.GuavaModule;
+import io.digdag.client.api.JacksonTimeModule;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+import static java.util.stream.Collectors.toList;
+
+public class CollectionPrinter<T>
+{
+    private static final ObjectMapper MAPPER = new ObjectMapper()
+            .registerModule(new GuavaModule())
+            .registerModule(new JacksonTimeModule());
+
+    private static final ObjectWriter YAML_WRITER = new ObjectMapper(new YAMLFactory()
+            .configure(YAMLGenerator.Feature.WRITE_DOC_START_MARKER, false))
+            .registerModule(new GuavaModule())
+            .registerModule(new JacksonTimeModule())
+            .writer();
+
+    private final static ObjectWriter JSON_WRITER = MAPPER
+            .writerWithDefaultPrettyPrinter();
+
+    private final List<Column<T>> columns = new ArrayList<>();
+
+    public void column(String name, Function<T, String> accessor)
+    {
+        columns.add(new Column<>(name, accessor));
+    }
+
+    public void print(OutputFormat f, List<T> items, OutputStream out)
+            throws IOException
+    {
+        PrintStream ps = new PrintStream(out);
+        print(f, items, ps);
+        ps.flush();
+    }
+
+    private void print(OutputFormat f, List<T> items, PrintStream ps)
+            throws IOException
```

Comment:
I think `BufferedWriter + OutputStreamWriter` is better than `PrintStream` because `PrintStream` doesn't throw exception at all. Error is maybe important especially in case that output is redirected to a file and the disk is full.